src/cloudWatchLogs/registry/logStreamRegistry.ts

```diff
@@ -181,9 +190,40 @@ export class LogStreamRegistry {
         this.activeStreams.set(uriToKey(uri), stream)
     }
 
-    private getLog(uri: vscode.Uri): CloudWatchLogsData | undefined {
+    public getLog(uri: vscode.Uri): CloudWatchLogsData | undefined {
         return this.activeStreams.get(uriToKey(uri))
     }
+
+    public setTextEditor(uri: vscode.Uri, textEditor: vscode.TextEditor): void {
+        this.activeTextEditors.set(uriToKey(uri), textEditor)
+    }
+
+    public getTextEditor(uri: vscode.Uri): vscode.TextEditor | undefined {
+        return this.activeTextEditors.get(uriToKey(uri))
+    }
+
+    public hasTextEditor(uri: vscode.Uri): boolean {
+        return this.activeTextEditors.has(uriToKey(uri))
+    }
+
+    public async highlightDocument(uri: vscode.Uri): Promise<void> {
+        const textEditor = this.getTextEditor(uri)
+        const logData = this.getLog(uri)
+
+        if (!logData) {
+            throw new Error(`Missing log data in registry for uri key: ${uriToKey(uri)}. Unable to highlight`)
+        }
+
+        if (!textEditor) {
+            throw new Error(`Missing textEditor in registry for uri key: ${uriToKey(uri)}. Unable to highlight`)
+        }
+
+        if (logData.parameters.filterPattern) {
+            const highlighter = vscode.window.createTextEditorDecorationType({ backgroundColor: 'blue' })
```

Comment:
See if there's already a [theme color](https://code.visualstudio.com/api/references/theme-color) that you could use. I think you can pass in the string id like `list.focusHighlightForeground` directly but not sure.